### ISSUE TITLE
SC2154 false positive when declaring array members with "printf -v" without quotes

### ISSUE THREAD

**Issue by manfredsc** [koalaman/shellcheck#2767]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and found the following related bug: #896

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
printf -v 'a[1]' "%s" "33"
echo "${a[1]}"

printf -v "b[1]" "%s" "33"
echo "${b[1]}"

printf -v c[1] "%s" "33"
echo "${c[1]}"
```

#### Here's what shellcheck currently says:
```
Line 9	SC2154: c is referenced but not assigned.
```
#### Here's what I wanted or expected to see:

Third hunk should behave like former ones, no warning, as c[1] is assigned by printf command.
